### DISCUSSION TITLE
Extend open list avoiding conflicts

### DISCUSSION THREAD

**Discussion by b4nst** [cuelang/cue#860] [Q&A]
Is there a way to have this behavior working?

```cue
foo: {
  bar: "bar"
  l: [...int]
}

foo: l: l & [1, 2, ...]
foo: l: l & [3, ...]
```

```json
{
    "foo": {
        "bar": "bar",
        "l": [
            1,
            2,
            3
        ]
    }
}
```

It's failing rightfully because `1` and `3` are conflicting. I've tried with `list.Concat` but since I want to use the same list I end up on a cycling error (also expected). 

Currently my workaround is to use an embedded definition + field comprehension
```cue
foo: {
	bar: "bar"
	#l: {[string]: int}
	l: [ for k, v in #l {v}]
}
foo: #l: "1": 1
foo: #l: "2": 2
foo: #l: "3": 3
```

**Comment by myitcv** [comment]
Do you find yourself in this situation there are multiple "sites" where you want to insert values?

If so, consider using hidden fields/definitions at each "site" and do something like:

```
package x

x: [
	for v in _a {v},
	for v in _b {v},
	for v in _c {v},
]

_a: [1, 2]
_b: [3]
_c: [8]
```

**Reply by b4nst** [reply]
Indeed this workaround works too, thanks!

**Comment by mpvl** [comment]
The idea is that this will be addressed with associative lists, See #165, but this is not implemented yet.

**Reply by b4nst** [reply]
Thanks!

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/860.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
